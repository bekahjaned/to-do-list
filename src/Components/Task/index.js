import React from "react";

import { Button } from "../../Elements/Button/";
import { TaskWrapper } from "../../Elements/TaskWrapper/";

class Task extends React.Component {
  render() {
    return (
      <TaskWrapper>
        <p>{this.props.text}</p>
        <Button onClick={this.props.deleteMethod}>x</Button>
      </TaskWrapper>
    );
  }
}
export default Task;
