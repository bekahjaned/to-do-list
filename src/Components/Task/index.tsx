import React, { MouseEventHandler } from "react";

import { Button } from "../../Elements/Button/";
import { TaskWrapper } from "../../Elements/TaskWrapper/";

type Props = {
  deleteMethod: MouseEventHandler;
  text: string;
};

const Task = ({ deleteMethod, text }: Props) => {
  return (
    <TaskWrapper>
      <p>{text}</p>
      <Button onClick={deleteMethod}>x</Button>
    </TaskWrapper>
  );
};
export default Task;
