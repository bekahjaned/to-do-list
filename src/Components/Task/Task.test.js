import React from "react";
import renderer from "react-test-renderer";

import Task from "./index";

it("renders a task", () => {
  // given
  const task = "wash dishes";

  //   when
  const tree = renderer.create(<Task text={task} />).toJSON();

  //   then
  expect(tree).toMatchSnapshot();
});
