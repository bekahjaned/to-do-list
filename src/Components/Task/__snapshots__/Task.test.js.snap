// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a task 1`] = `
<div
  className="sc-AxirZ glbUEU"
>
  <p>
    wash dishes
  </p>
  <button
    className="sc-AxjAm kuVXIB"
  >
    x
  </button>
</div>
`;

exports[`renders multiple tasks 1`] = `
<div
  className="sc-AxirZ glbUEU"
>
  <p>
    do this
    do that
  </p>
  <button
    className="sc-AxjAm kuVXIB"
  >
    x
  </button>
</div>
`;
